lib/chore/strategies/worker/single_worker_strategy.rb

```diff
@@ -25,16 +33,14 @@ def stop!
       # single worker strategy, this should never be called if the worker is in
       # progress.
       def assign(work)
-        if workers_available?
-          begin
-            @worker = worker_klass.new(work, @options)
-            @worker.start
-            true
-          ensure
-            @worker = nil
-          end
-        else
-          Chore.logger.error { "#{self.class}#assign: single worker is unavailable, but assign has been re-entered: #{caller * "\n"}" }
+        return unless acquire_worker
```

Comment:
Might be a brain fart on my end, one sec...

On Mon, Feb 1, 2016 at 1:09 PM, Aaron Pfeifer <notifications@github.com>
wrote:

> In lib/chore/strategies/worker/single_worker_strategy.rb
> <https://github.com/Tapjoy/chore/pull/25#discussion_r51455440>:
>
> > @@ -25,16 +33,14 @@ def stop!
> >        # single worker strategy, this should never be called if the worker is in
> >        # progress.
> >        def assign(work)
> > -        if workers_available?
> > -          begin
> > -            @worker = worker_klass.new(work, @options)
> > -            @worker.start
> > -            true
> > -          ensure
> > -            @worker = nil
> > -          end
> > -        else
> > -          Chore.logger.error { "#{self.class}#assign: single worker is unavailable, but assign has been re-entered: #{caller * "\n"}" }
> > +        return unless acquire_worker
>
> Not sure I follow -- as far as I can tell, this is what the forked code
> does (I effectively copied it from there). Mind pointing to the lines of
> code (in the forked worker) you're thinking about?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Tapjoy/chore/pull/25/files#r51455440>.
>